includes/discovery/sensors/humidity/webmon.inc.php

```diff
@@ -0,0 +1,31 @@
+<?php
+/*
+ * LibreNMS Dantel Webmon humidity sensor
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Mike Williams
+ * @author     Mike Williams <mike@mgww.net>
+ */
+
+$env = snmpwalk_group($device, 'pSlot1Table', 'WEBMON-EDGE-MATRIX-MIB');
+
+if (is_array($env)) {
```

Comment:
This sensor looks like it could be implemented in yaml, if so, please move it there.

If you keep in in PHP: snmpwalk_group always returns an array. You should check if it is empty instead.

Reply:
Fixed. The SNMP implementation is weird and I couldn't get it to pull the limits properly using YAML.